lib/concurrent/collection/array.rb

```diff
@@ -0,0 +1,38 @@
+require 'concurrent/utility/engine'
+require 'concurrent/thread_safe/util'
+
+module Concurrent
+  if Concurrent.on_cruby?
+
+    # Because MRI never runs code in parallel, the existing
+    # non-thread-safe structures should usually work fine.
+
+    # @!macro [attach] concurrent_array
+    #
+    #   A thread-safe subclass of Array. This version locks against the object
+    #   itself for every method call, ensuring only one thread can be reading
+    #   or writing at a time. This includes iteration methods like `#each`.
+    #
+    #   @see http://ruby-doc.org/core-2.2.0/Array.html Ruby standard library `Array`
+    class Array < ::Array;
```

Comment:
Ah sorry I got that confused, I thought we sticked with the previous name-spacing without aligning directories and modules. I am for alignment. The number of collections has quite a potential for growth so to avoid too many classes in Concurrent namespace I wanted to move them from the start. I'll move it back.

I am sorry I know we've discussed our name-spacing policy in the past but I do not remember clearly what the conclusion was. 

1) Any implementation or utility classes should not be in main space. Main space only contains Public classes. Same could be used for requirable files, only `concurrent/*` are intended for requiring by users. Clear, strict, but bigger main namespace.

2) also use sub spaces and use private constants to hide things. Requirable files are those which are documented as such. Less clear, more flexibility, less things in main space, more work with maintaining because of the fuzzier rules.

Let me know what you think and I'll update readme and also synchronisation layer which ended up non-standard.